pkg/relayer/proxy/server_builder.go

```diff
@@ -0,0 +1,62 @@
+package proxy
+
+import (
+	"context"
+
+	sharedtypes "pocket/x/shared/types"
+	suppliertypes "pocket/x/supplier/types"
+)
+
+// BuildProvidedServices builds the advertised relay servers from the supplier's on-chain advertised services.
+// It populates the relayerProxy's `advertisedRelayServers` map of servers for each service, where each server
+// is responsible for listening for incoming relay requests and relaying them to the supported native service.
```

Comment:
```suggestion
// is responsible for listening for incoming relay requests and relaying them to the supported proxied service.
```

I feel like `native` is a new term, but this whole package is called `proxy`